### PR TITLE
vpp-manager: rm phy routes&addr before vpp

### PR DESCRIPTION
**Pull request by sknat**
Signed-off-by: Nathan Skrzypczak <nathan.skrzypczak@gmail.com>

### CODE REVIEW

File: vpp-manager/uplink/common.go [hunk 1]
```diff
@@ -55,6 +55,36 @@ func (d *UplinkDriverData) GetName() string {
...
+		for _, addr := range d.conf.Addresses {
+			err := netlink.AddrDel(link, &addr)
+			if err != nil {
+				log.Errorf("Error adding address %s to tap interface : %+v", addr, err)
+			}
+		}
+
+		if d.conf.IsUp && setIfDown {
+			err = netlink.LinkSetDown(link)
+			if err != nil {
```
Comment (AloysAugustin) [inline vpp-manager/uplink/common.go:81]: In SafeSetInterfaceDownByName , if LinkSetDown errored we attempted to put the link back up, should we do it here as well?

File: vpp-manager/uplink/common.go [hunk 2]
```diff
@@ -55,6 +55,36 @@ func (d *UplinkDriverData) GetName() string {
...
+			err := netlink.RouteDel(&route)
+			if err != nil {
+				log.Errorf("cannot delete route %+v: %+v", route, err)
+			}
+		}
+		// Remove addresses to not have them conflict with vpptap0
+		for _, addr := range d.conf.Addresses {
+			err := netlink.AddrDel(link, &addr)
+			if err != nil {
+				log.Errorf("Error adding address %s to tap interface : %+v", addr, err)
```
Comment (AloysAugustin) [inline vpp-manager/uplink/common.go:75]: The log should be changed (adding -> removing)

File: vpp-manager/uplink/common.go (hunk 1 above)
Comment (sknat) [reply vpp-manager/uplink/common.go:81]: Right, yesterday I didn't recall why we did have it. But on a second thought it makes sense

File: vpp-manager/uplink/common.go (hunk 2 above)
Comment (sknat) [reply vpp-manager/uplink/common.go]: right

**Review by AloysAugustin — APPROVED** [review]

**Pull request merged**